css-size-adjust-1/Overview.bs

```diff
@@ -97,54 +108,90 @@ The default size adjustment affects text and form controls, whether
 those form controls contain text (e.g., text inputs, selects) or do not
 (e.g., radio buttons, checkboxes).
 
-Conditions that suppress adjustment {#default-adjustment-conditions}
---------------------------------------------------------------------
+Adjustment Conditions {#default-adjustment-conditions}
+------------------------------------------------------
+
+A User Agent may consider any information when determining which elements
+get adjusted.
 
-A number of conditions suppress the default adjustment because these
-conditions are associated with layouts for which the user experience
-would be worsened by size adjustment rather than improved by it.  These
-conditions are:
+Note: In general, User Agents are trying to determine which elements are
+integral to the structure of the page, and thus shouldn't be adjusted,
+and which elements are allowed to flow freely and thus can be adjusted
+without undesirable rendering artifacts.
 
+Here are some criteria which User Agents may consider when making this
+determination:
 <ul>
-  <li>when the total amount of text in the block formatting context (see
+  <li>When the total amount of text in the block formatting context (see
   [[!CSS21]]) (excluding text inside descendant block formatting
   contexts) is approximately smaller than the amount that would require
   wrapping to more than one or two lines within that context's
-  width,</li>
+  width. This is a signal that text doesn't need to be autosized.</li>
 
-  <li>when the objects to be adjusted are inside a block-level or
+  <li>When the objects to be adjusted are inside a block-level or
   ''display: inline-block'' element with a 'height' other than
-  ''height/auto'' (see [[!CSS21]]),</li>
+  ''height/auto'' (see [[!CSS21]]). This is a signal that the elements
+  are structural to the page content.</li>
 
-  <li>when the objects to be adjusted are inside a
+  <li>When the objects to be adjusted are inside a
   ''display: inline-block'' element
   with a 'width' other than ''width/auto'' (see
-  [[!CSS21]]),</li>
+  [[!CSS21]]). This is a signal that the elements
+  are structural to the page content.</li>
 
-  <li>when the objects to be adjusted have 'white-space' of ''white-space/pre'' or
+  <li>When the objects to be adjusted have 'white-space' of ''white-space/pre'' or
   ''white-space/nowrap'' (see [[!CSS21]]) or a 'text-wrap' of ''text-wrap/nowrap'' (see
   [[!CSS-TEXT-4]]).</li>
 
+  <li>Editability of an element, either because it's a form element or because it is ''contenteditable''</li>
+
+  <li>Element contents; empty elements might not benefit from being autosized</li>
+
+  <li>Elements which have compute styles that are fairly unique among the page. These are often structural
+  and might provide more broken layout than they benefit from size adjustment</li>
+
+  <li>Elements lay out to be extremely wide compared to the viewport. These would cause horizontal scrolling
```

Comment:
probably something like "lay out to be" "whose width is"

Reply:
That said, in Gecko we don't consider this a condition for disabling adjustment -- instead, we cap the width used to min(viewport inline-size, element inline-size) in [this code](https://searchfox.org/mozilla-central/rev/9b4b41b95cbcda63f565bdc24411e15248f91d83/layout/generic/nsFontInflationData.cpp#209-225).  So it's perhaps worth mentioning that option as well?